### ISSUE TITLE
[BUG] E0283: type annotations required in amethyst_assets/prefab/mod.rs

### ISSUE THREAD

**Issue by juliandemille** [amethyst/amethyst#1707]
## Description

Upon building, an E0283 is raised in amethyst_assets/src/prefab/mod.rs

## Reproduction Steps

1. Try to build Amethyst

## What You Expected to Happen

The build completes

## What Actually Happened

The build fails

## Screenshots and Logs

```
error[E0283]: type annotations required: cannot resolve `std::string::String: std::convert::AsRef<_>`
   --> /Users/julian/.cargo/registry/src/github.com-1ecc6299db9ec823/amethyst_assets-0.6.0/src/prefab/mod.rs:372:22
    |
372 |                 name.as_ref(),
    |
```

## Your Environment

1. Operating System: macOS
2. Operating System Version: Mojave
3. Version of Amethyst: 0.10.0

**Comment by Frizi** [comment]
Can you include rustc compiler version?

**Comment by torkleyy** [comment]
This is caused by a regression in nightly which won't be fixed.

**Comment by torkleyy** [comment]
https://github.com/rust-lang/rust/issues/60958#issuecomment-498084664

**Comment by distransient** [comment]
This should be fixed in 0.11.
